weblate/screenshots/models.py

```diff
@@ -76,3 +90,51 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    repository = component.repository
+    last_remote_revision = repository.last_remote_revision
+    changed_files = repository.get_changed_files(compare_to=previous_head)
+    screenshots = Screenshot.objects.filter(
+        translation__component=component, repository_filename__in=changed_files
+    )
+
+    # Update existing screenshots
+    for screenshot in screenshots:
+        filename = screenshot.repository_filename
+        changed_files.remove(filename)
+
+        file_data = repository.get_file(filename, last_remote_revision, raw=True)
+        image_file = File(file_data, name=filename)
+
+        try:
+            validate_bitmap(image_file)
+        except ValidationError:
+            report_error(cause="Failed to validate image from repository")
+            continue
+
+        screenshot.image = image_file
+        screenshot.save(update_fields=["image"])
+
+    # Update new screenshots matching screenshot filemask
+    for filename in changed_files:
+        if fnmatch.fnmatch(filename, component.screenshot_filemask):
+            file_data = repository.get_file(filename, last_remote_revision, raw=True)
+            image_file = File(file_data, name=filename)
+
+            try:
+                validate_bitmap(image_file)
+            except ValidationError:
+                report_error(cause="Failed to validate image from repository")
+                continue
+
+            screenshot = Screenshot.objects.create(
+                name=filename,
+                repository_filename=filename,
+                image=image_file,
```

Comment:
Same question here: Will the screenshot be stored in `screenshots` or kept in VCS location?

Reply:
<img width="1351" alt="Screenshot 2023-04-03 at 22 36 51" src="https://user-images.githubusercontent.com/24358501/229621991-ff90207a-5c2e-4e65-8606-a8d1b09454f5.png">


This is what I see in the screenshots table, It means the screenshot is stored in `screenshots`.

Reply:
Okay, that looks good.